sc2ts/utils.py

```diff
@@ -558,6 +559,107 @@ def get_imputed_pango(u):
+        meta = self.nodes_metadata
+        for u in self.recombinants:
+            md = meta[u]
+            row = md["match_info"]
+            assert len(row) == 2
+            assert row[0]["strain"] == row[1]["strain"]
+            matches = {}
+            for record in row:
+                matches[record["direction"]] = Match(
+                    breakpoints=record["breakpoints"],
+                    parents=record["parents"],
+                    mutations=record["mutations"],
+                )
+            rec = Recombinant(row[0]["strain"], matches, node=u)
+            breaks = rec.matches["forward"].breakpoints[1:-1]
+            for i in range(len(breaks)):
+                forward_breakpoints[breaks[i]].append(
+                    (u, rec.is_hmm_consistent(), rec.matches["forward"].parents[i:i+2])
+                )
+        sorted_breakpoints = sorted(forward_breakpoints.keys())
+        tree_a_indexes = np.searchsorted(
+            self.ts.breakpoints(as_array=True),
+            sorted_breakpoints,
+            side='left',
+        ) - 1
+        tree_a_indexes = np.searchsorted(
+            self.ts.breakpoints(as_array=True),
+            sorted_breakpoints,
+            side='right',
+        ) - 1
+
+        causal_sample_map = get_recombinant_samples(self.ts)
+        tree_a = self.ts.first()    
+        tree_b = self.ts.first()
+        node_times = self.ts.nodes_time
+        data = []
+        for i_a, i_b, brk in zip(tree_a_indexes, tree_a_indexes, sorted_breakpoints):
+            tree_a.seek(i_a)
+            tree_b.seek(i_b)
+            for nd, hmm_cons, (left_parent, right_parent) in forward_breakpoints[brk]:
+                num_nodes_in_path = 0
+                node_a = left_parent
+                node_b = right_parent
+                while True:
+                    if node_a == tskit.NULL or node_b == tskit.NULL:
+                        break
+                    if node_times[node_a] == node_times[node_b]:
+                        if node_a == node_b:
+                            break
+                        else:
+                            if tree_a.parent(node_a) < tree_b.parent(node_b):
+                                node_a = tree_a.parent(node_a)
+                            else:
+                                node_b = tree_b.parent(node_b)
+                            num_nodes_in_path += 1
+                                
+                    elif node_times[node_a] < node_times[node_b]:
+                        node_a = tree_a.parent(node_a)
+                        num_nodes_in_path += 1
+                    elif node_times[node_b] < node_times[node_a]:
+                        node_b = tree_b.parent(node_b)
+                        num_nodes_in_path += 1
+                mrca = node_a
+                row = {
+                    "node": nd,
+                    "breakpoint": brk,
+                    "HMM_consistent": hmm_cons,
```

Comment:
Might as well be consistent about keeping things lowercase with `hmm_consistent``. MRCA is also an initialism , and we've lowercased this